components/Portfolio/Trading/Overview/index.tsx

```diff
@@ -21,17 +21,19 @@ import { LabelledTracers } from '@libs/types/TracerTypes';
 
 interface OProps {
     tracers: LabelledTracers;
-    positions: Tracer[];
-    holdings: Tracer[];
+    fetchedTracers: Tracer[];
     allFilledOrders: LabelledOrders;
 }
-const Overview: FC<OProps> = ({ tracers, positions, holdings, allFilledOrders }: OProps) => {
+const Overview: FC<OProps> = ({ tracers, fetchedTracers, allFilledOrders }: OProps) => {
     const { account } = useWeb3();
     const [currentPortfolio, setCurrentPortfolio] = useState(-1);
     const [currentPNL, setCurrentPNL] = useState(1);
 
     const portfolioKeyMap: Record<number, string> = {};
     portfolioKeyMap[-1] = 'Entire Portfolio';
+
+    const positions: Tracer[] = fetchedTracers?.filter((tracer) => !tracer.getBalance().base.eq(0));
+    const holdings: Tracer[] = fetchedTracers?.filter((tracer) => !tracer.getBalance().quote.eq(0));
```

Comment:
This will result in a fair bit of unnecessary computation. Id probably move it into a useState with an effect

Reply:
mmm, useState in an effect was part of what caused the refresh issue